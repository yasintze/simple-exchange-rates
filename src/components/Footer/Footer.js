// @flow
import React from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";

type Props = {
  date: string
};

const Footer = (props: Props) => {
  const { date } = props;
  const lastUpdatedInfo = `Rates last updated ${date}`;

  return (
    <Grid container>
      <Grid item xs>
        <Typography align="center" variant="overline" gutterBottom>
          {date && <strong>{lastUpdatedInfo}</strong>}
          <br />
          {"by "}
          <a href="https://github.com/yasintze/simple-exchange-rates">Yasin</a>
        </Typography>
      </Grid>
    </Grid>
  );
};

export default Footer;
