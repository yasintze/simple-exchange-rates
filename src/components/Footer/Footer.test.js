// @flow
import React from "react";
import { shallow } from "enzyme";

import Footer from "./Footer";

describe("Footer component", () => {
  it("render correctly", () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });

  it("contain 'strong' tag", () => {
    const wrapper = shallow(<Footer date="2019-02-16" />);
    expect(wrapper.find("strong").exists()).toBe(true);
  });
});
