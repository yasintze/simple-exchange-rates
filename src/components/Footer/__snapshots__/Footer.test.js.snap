// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Grid)
        item={true}
        xs={true}
      >
        <WithStyles(Typography)
          align="center"
          gutterBottom={true}
          variant="overline"
        >
          <br />
          by
          <a
            href="https://github.com/yasintze/simple-exchange-rates"
          >
            Yasin Junet
          </a>
        </WithStyles(Typography)>
      </WithStyles(Grid)>,
      "container": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Typography)
          align="center"
          gutterBottom={true}
          variant="overline"
        >
          <br />
          by
          <a
            href="https://github.com/yasintze/simple-exchange-rates"
          >
            Yasin Junet
          </a>
        </WithStyles(Typography)>,
        "item": true,
        "xs": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "children": Array [
            undefined,
            <br />,
            "by",
            <a
              href="https://github.com/yasintze/simple-exchange-rates"
            >
              Yasin Junet
            </a>,
          ],
          "gutterBottom": true,
          "variant": "overline",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "by",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Yasin Junet",
              "href": "https://github.com/yasintze/simple-exchange-rates",
            },
            "ref": null,
            "rendered": "Yasin Junet",
            "type": "a",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Grid)
          item={true}
          xs={true}
        >
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="overline"
          >
            <br />
            by
            <a
              href="https://github.com/yasintze/simple-exchange-rates"
            >
              Yasin Junet
            </a>
          </WithStyles(Typography)>
        </WithStyles(Grid)>,
        "container": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="overline"
          >
            <br />
            by
            <a
              href="https://github.com/yasintze/simple-exchange-rates"
            >
              Yasin Junet
            </a>
          </WithStyles(Typography)>,
          "item": true,
          "xs": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "center",
            "children": Array [
              undefined,
              <br />,
              "by",
              <a
                href="https://github.com/yasintze/simple-exchange-rates"
              >
                Yasin Junet
              </a>,
            ],
            "gutterBottom": true,
            "variant": "overline",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "by",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Yasin Junet",
                "href": "https://github.com/yasintze/simple-exchange-rates",
              },
              "ref": null,
              "rendered": "Yasin Junet",
              "type": "a",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
